### ISSUE TITLE
Ktlint reporting unexpected indentation when formatting lambdas with trailing comma

### ISSUE THREAD

**Issue by matejdro** [pinterest/ktlint#1247]
Following code:

```kotlin
fun test() {
    lateinit var a: (String, String, String) -> String

    a = {
            s: String,
            s1: String,
            s2: String,
        ->
        TODO()
    }
}
```

has been formatted with default IntelliJ kotlin code style + trailing comma enabled

## Expected Behavior

ktlint should accept the code

## Observed Behavior

ktlint fails with 

```
Test.kt:5:1 Unexpected indentation (12) (should be 8) (indent)
Test.kt:6:1 Unexpected indentation (12) (should be 8) (indent)
Test.kt:7:1 Unexpected indentation (12) (should be 8) (indent)
```

## Your Environment

* Version of ktlint used: 0.42.1
* Name and version (or code for custom task) of integration used (Gradle plugin, Maven plugin, command line, custom Gradle task): ktlint-gradle 10.2.0

**Comment by Kernald** [comment]
This also seems to not be playing well with #1032 at all.

**Comment by paul-dingemans** [comment]
This is really an interesting code sample. The trailing comma after variable `s2` makes a huge difference for the formatting.

When code style preference `Use trailing comma` in IntelliJ is disabled and `s2` is not followed by a trailing comma , the code gets formatted as:
```
fun test() {
    lateinit var a: (String, String, String) -> String

    a = { s: String,
          s1: String,
          s2: String
        ->
        TODO()
    }
}
```

But in both cases the IntelliJ formatting looks better than the current formatting of Ktlint shown below:
```
fun test() {
    lateinit var a: (String, String, String) -> String

    a = {
        s: String,
        s1: String,
        s2: String
        ->
        TODO()
    }
}
```

I think we should use the same formatting style in both cases. I prefer the style which is used when no trailing comma is found, so:
```
fun test() {
    lateinit var a: (String, String, String) -> String

    a = { s: String, 
          s1: String,
          s2: String
        ->
        TODO()
    }
}
```
@romtsn @Tapchicoma What do you think?

**Comment by paul-dingemans** [comment]
I prepared a fix but have to come back on my preference above. This looks prettier in this example due to the extremely short variable name `a`. With longer variable names, it is better to stick to format:
```
fun test() {
    lateinit var a: (String, String, String) -> String

    a = {
            s: String,
            s1: String,
            s2: String,
        ->
        TODO()
    }
}
```

**Comment by matejdro** [comment]
Can we even decide on the format, though? I assume whatever IntelliJ's formatter spits out, it must also be valid ktlint formatting? Since, from what I can see, there is no way to tell IntelliJ's formatter to format it like that.

**Comment by paul-dingemans** [comment]
For me the only requirements are:
* Output of ktlint should be accepted by IntelliJ formatting rules
* ktlint should format consistently.

Code formatted by fix #1293 applies to those rules.

**Comment by matejdro** [comment]
Sorry, I'm a bit confused here. Does that mean that above code, formatted by IntelliJ, will not be red anymore? Or do we need to format it with ktlint every time and disable/ignore IntelliJ's formatter?
